### ISSUE TITLE
TSP code called with shipments

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#333]
The old TSP code-base [is still called](https://github.com/VROOM-Project/vroom/blob/c45754eef74b5e7537b718314a4c3d0795be0245/src/problems/cvrp/cvrp.cpp#L143-L155) when we encounter a specific case of CVRP that has only one vehicle, no skills and no amount.

This was fine when we only had `jobs` in input. But now when inputting a problem with `shipments` (and only one vehicle, no skills and no amount), then it's solved as a plain TSP so we don't guarantee the precedence constraints for `shipments` tasks. This results in hitting various asserts or simply crashing with `NDEBUG`.

One might argue that having `shipments` without `amounts` sounds weird, but the current code-base does support this (when solving appropriately) and it's a totally valid way of modelling same-route precedence constraints without any capacity constraint.

**Comment by jcoupey** [comment]
This is actually a one-line fix:

```patch
diff --git a/src/problems/cvrp/cvrp.cpp b/src/problems/cvrp/cvrp.cpp
index 1323daa..5d03f1f 100644
--- a/src/problems/cvrp/cvrp.cpp
+++ b/src/problems/cvrp/cvrp.cpp
@@ -141,7 +141,7 @@ Solution CVRP::solve(unsigned exploration_level,
                      unsigned nb_threads,
                      const std::vector<HeuristicParameters>& h_param) const {
   if (_input.vehicles.size() == 1 and !_input.has_skills() and
-      _input.zero_amount().size() == 0) {
+      _input.zero_amount().size() == 0 and !_input.has_shipments()) {
     // This is a plain TSP, no need to go through the trouble below.
     std::vector<Index> job_ranks(_input.jobs.size());
     std::iota(job_ranks.begin(), job_ranks.end(), 0);
```

So I don't think it's worth doing a patched `v1.6.1` release. I'll just let this ticket open for reference until `v1.7.0` is released.

**Comment by jcoupey** [comment]
Fixed in `v1.7.0`.
